tests/unit/algos/list/delimiterMaster.spec.js

```diff
@@ -46,6 +48,116 @@ function getListingKey(key, vFormat) {
+ * @returns 
+ */
+function handleStreak(filteringResult, skippingRange, state) {
+    if (filteringResult < 0) {
+        // readStream would be destroyed. In this mock, we just continue.
+    } else if (filteringResult === 0 && skippingRange) {
+        // check if MAX_STREAK_LENGTH consecutive keys have been
+        // skipped
+        if (++state.streakLength === MAX_STREAK_LENGTH) {
+            if (Array.isArray(skippingRange)) {
+                // With synchronized listing, the listing
+                // algo backend skipping() function must
+                // return as many skip keys as listing
+                // param sets.
+                for (let i = 0; i < skippingRange.length; ++i) {
+                    // eslint-disable-next-line no-param-reassign
+                    state.params[i].gte = inc(skippingRange[i]);
+                }
+            } else {
+                // eslint-disable-next-line no-param-reassign
+                
+                state.params.gte = inc(skippingRange);
+            }
+            if (state.gteparams && state.gteparams === state.params.gte) {
+                streakLength = 1;
+            } else {
+                // stop listing this key range
+                // eslint-disable-next-line no-param-reassign
+                state.gteparams = state.params.gte;
+            }
+        }
+    } else {
+        state.streakLength = 1;
+    }
+}
+
+/**
+ * Generate a random number of versioned keys.
+ * @param {Integer} maxKeys - how many versioned keys to generate
+ * @yields {Object} - { key, isDeleteMarker }
+ */
+function *generateVersionedKeys(masterKey, numKeys) {
+    for (let i = 0; i < numKeys; i++) {
+        yield {
+            key: `${masterKey}${VID_SEP}${zpad(i)}`, 
+            isDeleteMarker: chance.bool(),
+            canSkip: true
+        }
+    }
+}
+
+/**
+ * Generate raw listing keys in alphabetical order.
+ * @param {Integer} count - how many keys to generate.
+ * @param {Object} state - state for test run.
+ * @yields {Object} - { key, isDeleteMarker }
+ */
+
+function *generateKeys(count, state) {
+    let idx = 0;
+    let masterKey = '_Common-Prefix/';
+    yield { key: masterKey, isDeleteMarker: false, canSkip: false }
+
+    idx++;
+    while (idx < count) {
+        masterKey = `_Common-Prefix/${zpad(idx)}`;
+        const masterKeyWithSubprefix = `_Common-Prefix/${zpad(idx)}/${zpad(idx)}`
+
+        const hasSubprefix = chance.bool();
+        const isDeleteMarker = chance.bool();
+        const baseKey = hasSubprefix ? masterKeyWithSubprefix : masterKey;
+        let canSkip = isDeleteMarker;
+
+        if (hasSubprefix) {
+            // 'prefix -> liof = -1, pin = 5'
+            // 'prefix/ -> liof = 6, pin = 6'
+            canSkip = state.handleSubprefixKey(baseKey, canSkip)
+        }
+
+        if (state.prefix[state.prefix.length - 1] !== '/') {
+            canSkip = state.handleSubprefixKey(baseKey, canSkip)
+        }
```

Comment:
```suggestion
        if (hasSubprefix || state.prefix[state.prefix.length - 1] !== '/') {
            canSkip = state.handleSubprefixKey(baseKey, canSkip);
        }
```

Also from a style perspective, I'd rather avoid passing `canSkip` to the helper function which should only deal with what it's named for, i.e. subprefix handling, otherwise it becomes hard to follow the logic, we can always use boolean operators to cope with this, something like `canSkip = isDeleteMarker || state.handleSubprefixKey(baseKey)` would likely work.